crates/tools/api/src/main.rs

```diff
@@ -1,56 +1,44 @@
-use std::io::Write;
-use windows_metadata::reader::{File, Reader, Tree};
+use std::{
+    fs,
+    io::Write,
+    path::PathBuf,
+    process::{Command, Stdio},
+};
+use windows_metadata::reader::File;
 
 fn main() {
-    let start = std::time::Instant::now();
-    let mut output_path = std::path::PathBuf::from("");
+    let files = [
+        File::new(".windows/winmd/Microsoft.Dia.winmd").unwrap(),
+        File::new(".windows/winmd/Windows.Win32.winmd").unwrap(),
+        File::new(".windows/winmd/Windows.Win32.Interop.winmd").unwrap(),
+    ];
 
-    output_path.push("src/Microsoft");
-    let _ = std::fs::remove_dir_all(&output_path);
-    output_path.pop();
+    let output_path = PathBuf::from("src/Microsoft/Dia");
+    fs::remove_dir_all(&output_path).unwrap();
+    fs::create_dir_all(&output_path).unwrap();
 
-    let winmd_files = [File::new(".windows/winmd/Microsoft.Dia.winmd").unwrap(), File::new(".windows/winmd/Windows.Win32.winmd").unwrap(), File::new(".windows/winmd/Windows.Win32.Interop.winmd").unwrap()];
-    let reader = Reader::new(&winmd_files);
-    let root = reader.tree("Microsoft", &[]).expect("Microsoft namespace not found");
+    let mut generated_tokens = windows_bindgen::component("Microsoft.Dia", &files);
```

Comment:
This could conceivably just go directly into a `src/bindingsrs`

Reply:
And maintain the `Microsoft::Dia` namespace? (See also https://github.com/microsoft/dia-rs/pull/10#discussion_r1042587972)

Reply:
I see now, will simplify that, thanks!